integration/tests/quickstart_profiles/quickstart_profiles_test.go

```diff
@@ -0,0 +1,110 @@
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var params *tests.Params
+
+func init() {
+	// Call testing.Init() prior to tests.NewParams(), as otherwise -test.* will not be recognised. See also: https://golang.org/doc/go1.13#testing
+	testing.Init()
+	params = tests.NewParams("qstart")
+}
+
+func TestSuite(t *testing.T) {
+	testutils.RegisterAndRun(t)
+}
+
+var _ = BeforeSuite(func() {
+	if params.DoCreate {
+		cmd := params.EksctlCreateCmd.WithArgs(
+			"cluster",
+			"--name", params.ClusterName,
+			"--verbose", "4",
+			"--region", params.Region,
+			"--kubeconfig", params.KubeconfigPath,
+		)
+		Expect(cmd).To(RunSuccessfully())
+	}
+})
+
+var _ = Describe("Enable and use GitOps quickstart profiles", func() {
+	var (
+		branch   string
+		cloneDir string
+		err      error
+	)
+
+	BeforeEach(func() {
+		if branch == "" {
+			branch = namer.RandomName()
+			cloneDir, err = git.CreateBranch(branch, params.PrivateSSHKeyPath)
+		}
+	})
+
+	Context("enable repo", func() {
+		It("should add Flux to the repo and the cluster", func() {
+			Expect(err).NotTo(HaveOccurred()) // Creating the branch should have succeeded.
+			AssertFluxManifestsAbsentInGit(branch, params.PrivateSSHKeyPath)
+			AssertFluxPodsAbsentInKubernetes(params.KubeconfigPath)
+
+			cmd := params.EksctlExperimentalCmd.WithArgs(
+				"enable", "repo",
+				"--git-url", git.Repository,
+				"--git-email", git.Email,
+				"--git-private-ssh-key-path", params.PrivateSSHKeyPath,
+				"--git-branch", branch,
+				"--cluster", params.ClusterName,
+			)
+			Expect(cmd).To(RunSuccessfully())
+
+			AssertFluxManifestsPresentInGit(branch, params.PrivateSSHKeyPath)
+			AssertFluxPodsPresentInKubernetes(params.KubeconfigPath)
+		})
+	})
+
+	Context("enable profile", func() {
+		It("should add the configured quickstart profile to the repo and the cluster", func() {
+			Expect(err).NotTo(HaveOccurred()) // Creating the branch should have succeeded.
+			// Flux should have been installed by the previously run "enable repo" command:
+			AssertFluxManifestsPresentInGit(branch, params.PrivateSSHKeyPath)
+			AssertFluxPodsPresentInKubernetes(params.KubeconfigPath)
+
+			cmd := params.EksctlExperimentalCmd.WithArgs(
+				"enable", "profile",
+				"--git-url", git.Repository,
+				"--git-email", git.Email,
+				"--git-branch", branch,
+				"--git-private-ssh-key-path", params.PrivateSSHKeyPath,
+				"--cluster", params.ClusterName,
+				"app-dev",
+			)
+			Expect(cmd).To(RunSuccessfully())
+
+			AssertQuickStartComponentsPresentInGit(branch, params.PrivateSSHKeyPath)
+			// Flux should still be present:
+			AssertFluxManifestsPresentInGit(branch, params.PrivateSSHKeyPath)
+			AssertFluxPodsPresentInKubernetes(params.KubeconfigPath)
+			// Clean-up:
+			err := git.DeleteBranch(branch, cloneDir, params.PrivateSSHKeyPath)
+			Expect(err).NotTo(HaveOccurred()) // Deleting the branch should have succeeded.
+		})
+	})
+})
+
+var _ = AfterSuite(func() {
```

Comment:
I am curious, does this get called when one kills the tests with SIGTERM?

Reply:
Testing I did with dummy tests did show that, yes, as I propagate the signal to child `ginkgo` processes and then, these stop test suites and call `AfterSuite`:

> Since the AfterSuite typically includes code to clean up persistent state ginkgo will also run AfterSuite when you send the running test suite an interrupt signal (^C).

Source: https://onsi.github.io/ginkgo/#global-setup-and-teardown-beforesuite-and-aftersuite

Reply:
NICE!